oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -81,4 +100,179 @@ public void multiValuedBinarySize() throws Exception {
+        getCompressedValueMethod.setAccessible(true);
+        byte[] result = (byte[]) getCompressedValueMethod.invoke(state);
+
+        assertNull(result);
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+    }
+
+    @Test
+    public void stringAboveThresholdSizeNoCompression() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+
+        // Get the private method
+        Method getCompressedValueMethod = DocumentPropertyState.class.getDeclaredMethod("getCompressedValue");
+
+        // Make the method accessible
+        getCompressedValueMethod.setAccessible(true);
+
+        Field compressionThreshold = DocumentPropertyState.class.getDeclaredField("DEFAULT_COMPRESSION_THRESHOLD");
+        compressionThreshold.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(compressionThreshold, compressionThreshold.getModifiers() & ~Modifier.FINAL);
+
+        compressionThreshold.set(null, DEFAULT_COMPRESSION_THRESHOLD);
+
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", "\"" + STRING_HUGEVALUE + "\"", Compression.NONE);
+
+        byte[] result = (byte[]) getCompressedValueMethod.invoke(state);
+
+        assertEquals(result.length, STRING_HUGEVALUE.length() + 2 );
+
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+        assertEquals(STRING_HUGEVALUE, state.getValue(Type.STRING));
+
+        // Reset the value of the field
+        compressionThreshold.set(null, -1);
+    }
+
+    @Test
+    public void performanceTest() {
+        String testString = RandomStringUtils.random(10050, "dummytest");
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", "\"" + testString + "\"", Compression.GZIP);
+
+        Runtime runtime = Runtime.getRuntime();
+
+        long startMemory = runtime.totalMemory() - runtime.freeMemory(); // Get initial memory usage
+        long startTime = System.nanoTime();
+
+        String compressedValue = state.getValue(Type.STRING);
+
+        long endTime = System.nanoTime();
+        long endMemory = runtime.totalMemory() - runtime.freeMemory(); // Get final memory usage
+
+        long duration = (endTime - startTime);  // divide by 1000000 to get milliseconds.
+        long usedMemory = endMemory - startMemory; // Calculate the used memory during the operation
+
+        System.out.println("Time taken for compression: " + duration + " nanoseconds");
+        System.out.println("Memory used for compression: " + usedMemory + " bytes");
+
+        // Assert that the compressed value is not null or empty
+        assertTrue(compressedValue != null && !compressedValue.isEmpty());
+    }
+}
```

Comment:
We absolutely need to test string values which would fail with a pedantic UTF-8 encoder. See https://issues.apache.org/jira/browse/OAK-10803?focusedCommentId=17855529&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17855529.

Reply:
Done.

Reply:
IIRC, the test fails, right? That means that for certain property values, compression would be lossy, right?

Reply:
The test failed. The compression didn't work properly.

Reply:
(assuming this was originally referring to `testInterestingStrings`)

> The test failed. The compression didn't work properly.

So is the plan to still fix this? I don't think this test should be expected to throw `ComparisonFailure`

Reply:
Done

Reply:
[Commented](https://issues.apache.org/jira/browse/OAK-10803?focusedCommentId=17856243&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17856243) in the ticket - to me it looks like this `assertEquals(value,"");` is not what we'd aim for.

Reply:
The compression did work properly even for interesting strings. Done